### PR TITLE
DEV: Outlet for unobtrusive secondary full-page search.

### PR DESCRIPTION
**Pull request by romanrizzi**
`discourse-ai` will use the outlet to perform a parallel semantic search.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/controllers/full-page-search.js [hunk 1]
```diff
@@ -510,6 +511,9 @@ export default Controller.extend({
           ?.removeAttribute("open");
       }
       this.set("page", 1);
+
+      this.appEvents.trigger("full-page-search:trigger-search");
```
Comment (xfalcox) [inline app/assets/javascripts/discourse/app/controllers/full-page-search.js:515]: Not a problem, but I'm wondering why did you add a front-end hook instead of one in the backend :thinking:

**Review by xfalcox — APPROVED** [review]

File: app/assets/javascripts/discourse/app/controllers/full-page-search.js (hunk 1 above)
Comment (romanrizzi) [reply app/assets/javascripts/discourse/app/controllers/full-page-search.js:515]: We already had an endpoint for semantic search, so I decided to keep it instead of extending core in a different way.

Btw, this is the only way I found to trigger a new search when the core one happens. I couldn't make it work with the did-update modifier.
